Add context menu to reload button

Liunkae opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Fixes #5796

This enhancement takes advantage of `LongPressButton`, which has long press and right-click functionality out of the box. The context menu contains **Reload Page** and **Clean Reload**, the reload options currently implemented in Brave (from the View menu). Thus, Reload Page and Clean Reload have been moved to `CommonMenu`.

Test Plan:
1. Open a new Brave window.
2. Navigate to any site.
3. Right-click the reload button in the urlBar.
4. Make sure the context menu appears.
5. Click Reload Page.
6. Make sure the page is reloaded.
7. Right-click the reload button again.
8. Click Clean Reload.
9. Make sure the page is reloaded clean.
10. Long press the reload button.
11. Make sure the context menu also appears.

Screenshot (updated):
![image](https://cloud.githubusercontent.com/assets/19424103/20577150/a6205b0e-b186-11e6-99e7-6e60d6412c69.png)

bradleyrichter commented:
++

bsclifton commented:
Woah! Great job with this :heart: !!

Liunkae commented:
The context menu is now a native menu. New screenshot:
![image](https://cloud.githubusercontent.com/assets/19424103/20577150/a6205b0e-b186-11e6-99e7-6e60d6412c69.png)

bsclifton commented:
@Liunkae with a popup, this might cause an issue; what happens if you have a YouTube video playing and then the menu pops up?  I think it'll freeze the app :frowning_face:

Liunkae commented:
@bsclifton Yes, it does freeze temporarily. But do you think the benefits (appearance, long press drag release, staying in screen bounds, etc.) of a native menu outweigh the downside (temporary freeze).

bsclifton commented:
@bbondy @bradleyrichter what do you think of the above ^^ ?

Liunkae commented:
By the way, also using a native menu for the new tab dropdown could fix #5763.
![image](https://cloud.githubusercontent.com/assets/19424103/20579370/e46fc830-b191-11e6-9ad2-c4e4876745f8.png)

Liunkae commented:
Just in case, I'll leave the code to change back to non-native menus:
In `contextMenus.js`:
````js
function onReloadContextMenu (target) {
  const rect = target.getBoundingClientRect()
  const menuTemplate = [
    CommonMenu.reloadPageMenuItem(),
    CommonMenu.cleanReloadMenuItem()
  ]

  windowActions.setContextMenuDetail(Immutable.fromJS({
    left: rect.left,
    top: rect.bottom + 2,
    template: menuTemplate
  }))
}
````

bsclifton commented:
@Liunkae if you can change it back to a non-native menu for now and then rebase, that would be awesome :smile: 

We definitely appreciate the work and the native version does look good- but we'll want to keep the non-native version until we can solve (or work around)  the screen blocking during popup

bsclifton commented:
One advantage of the non-native menus (and feel free to take advantage of this):
You can test them using webdriver :smile: You should be able to find long press test cases for the back/forward buttons and also the new tab + (if you wanted to add one for this too)

Liunkae commented:
@bsclifton I changed back to a non-native menu and fixed the file conflicts.

Sorry for asking, but how do I rebase with the Brave master branch (since this branch is in my fork)? I searched everywhere, but I just cannot figure it out…

luixxiul commented:
git checkout reload-dropdown
git fetch upstream
git rebase upstream/master

and follow the instruction :-)

bsclifton commented:
@Liunkae no worries- I just rebased it for you! Let me give it one last test, then I'll merge. Thanks again :smile:

bsclifton commented:
Tested on macOS and Windows, looking great! :+1:
